src/class-processors.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Processors class file
+ *
+ * @package feed-consumer
+ */
+
+namespace Feed_Consumer;
+
+use Mantle\Support\Traits\Singleton;
+
+use function Mantle\Support\Helpers\collect;
+
+/**
+ * Processors Manager
+ */
+class Processors {
+	use Singleton;
+
+	/**
+	 * Processors.
+	 *
+	 * @var array
+	 */
+	protected array $processors = [
+		\Feed_Consumer\Processor\RSS_Processor::class,
+	];
+
+	/**
+	 * Constructor.
+	 */
+	protected function __construct() {
+		add_action( 'init', [ $this, 'on_init' ] );
+	}
+
+	/**
+	 * On init.
+	 */
+	public function on_init() {
+		/**
+		 * Filter the supported processor classes.
+		 *
+		 * @var string[]
+		 */
+		$this->processors = apply_filters( 'feed_consumer_processors', $this->processors );
+	}
+
+	/**
+	 * Retrieve or set the processors.
```

Comment:
Is using a single method vs. set and get methods preferable in your opinion? I could see this making the implementing code less clear. In this scenario, `$this->processors`, `$this->processors()`, and `$this->processors( 'something' )` are two different things with three different expected outcomes, and that feels a little confusing to me.

Reply:
That's a good point, I'm going to split the methods into a get/set pair. Will do in a follow up because I have a few PRs stack at the moment.